packages/fx-core/src/component/deploy/npmBuildDriver.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { checkMissingArgs } from "../utils/common";
+import { execute } from "../code/utils";
+import { ExecuteCommandError } from "../error/componentError";
+import { BaseBuildDriver } from "./baseBuildDriver";
+import { Service } from "typedi";
+import { StepDriver } from "../interface/stepDriver";
+import { DriverContext } from "../interface/buildAndDeployArgs";
+
+@Service("build/npm")
+export class NpmBuildDriver implements StepDriver {
+  async run(args: unknown, context: DriverContext): Promise<Map<string, string>> {
+    const impl = new NpmBuildDriverImpl(args, context);
+    return await impl.run();
+  }
+}
+
+export class NpmBuildDriverImpl extends BaseBuildDriver {
+  private static readonly emptyMap = new Map<string, string>();
+  private static readonly NPM_BUILD_COMMAND_PREFIX = "npm";
+
+  async run(): Promise<Map<string, string>> {
```

Comment:
It looks the run() looks similar between DotnetBuildDriver and NpmBuildDriver, is it possible to abstract more?

Reply:
Yes, or maybe it will both extend from the script driver. I'll make some changes when other people's code is ready